src/en/reference-constraints.md

```diff
@@ -109,33 +109,56 @@ multiple constraints delimited by a space.
     EC2 is the only provider supporting spaces constraints. Support for other
     providers is planned for future releases.
 
-## Legacy constraints
+- virt-type
 
-In pre-1.0 juju some additional or differently named constraints were
-also supported, these need to be migrated when upgrading.
+    Specifies the type of virtualization to be used, such as `kvm`.
 
-- cpu
 
-    Number of CPU cores for most providers, but equivalent to an Amazon
-    ECU on AWS. Use `cpu-cores` instead.
+## Provider differences
 
-- ec2-zone
+Different providers support different constraints and sometimes different
+values for these constraints. Sometimes, different providers also dictate
+constraints that would conflict with other providers and cannot be used
+in combination. Use this list to help you understand the differing needs.
 
-    EC2 availability zone that a service unit must be deployed into. No
-    equivalent implemented as of juju 1.12, follow [bug
-    1183831](https://bugs.launchpad.net/juju-core/+bug/1183831).
+###Azure Provider:
```

Comment:
This is better than not having the information, but it still feels rather clunky to use and understand. Maybe for now it is sufficient to add a TODO to the doc and file a new issue (medium priority)